### PR TITLE
Fix Admin Page Initialization for Extended Classes

### LINKED ISSUE #3443: EDTR - Fix creating custom message templates
Using Dev branch if you try to create a custom message template from within the event editor it fails.

To reproduce just click 'Create New Custom' on any message type in the notifications section of the event editor.

Give it a name and a description and click create. The ajax request will throw a fatal with this error:

```
[25-Jun-2021 09:09:53 UTC] PHP Fatal error:  Uncaught Error: Call to a member function get_messenger() on null in C:\laragon\www\ee4editor\wp-content\plugins\event-espresso-core\admin_pages\messages\Messages_Admin_Page.core.php:2734
Stack trace:
#0 C:\laragon\www\ee4editor\wp-content\plugins\event-espresso-core\admin_pages\messages\Messages_Admin_Page.core.php(1273): Messages_Admin_Page->_insert_or_update_message_template(true)
#1 C:\laragon\www\ee4editor\wp-content\plugins\event-espresso-core\admin_pages\messages\Messages_Admin_Page.core.php(1288): Messages_Admin_Page->_add_message_template('payment', 'email', '1')
#2 C:\laragon\www\ee4editor\wp-content\plugins\event-espresso-core\caffeinated\admin\extend\messages\espresso_events_Messages_Hooks_Extend.class.php(246): Messages_Admin_Page->add_message_template('payment', 'email', '1')
#3 C:\laragon\www\ee4editor\wp-includes\class-wp-hook.php(292): espresso_events_Messages_Hooks_Extend->create_new_custom('')
#4 C:\laragon\www\ee4editor\wp-includes\class-wp-hook.php(316): WP_Hook->apply_filters('', Array)
 in C:\laragon\www\ee4editor\wp-content\plugins\event-espresso-core\admin_pages\messages\Messages_Admin_Page.core.php on line 2734
```

### PR DESCRIPTION
**Pull request by tn3rb**
fixes #3443

Previously there was a chicken <--> egg scenario for child classes of `EE_Admin_Page` that needed to set properties prior to `EE_Admin_Page` running some of its initialization logic but also needed `EE_Admin_Page` to have its construction completed. To remedy this, the additional initialization logic was moved into `EE_Admin_Page::initializePage()`. 

Unfortunately I missed that child classes of `EE_Admin_Page` were also instantiated in `EE_Admin_Hooks::_set_page_object()` which led to the issue reported by @Pebblo in #3443.

This PR:

- calls `EE_Admin_Page::initializePage()` after constructing child classes
- adds a boolean `$initialized` prop to `EE_Admin_Page` so that the above logic only runs once
- ensures that `EE_Admin_Hooks::_set_page_object()` is only run once (by checking for that object)
- does some other minor code cleanup and formatting

### CODE REVIEW

File: core/helpers/EEH_Debug_Tools.helper.php [hunk 1]
```diff
@@ -587,8 +593,8 @@ public static function printr(
         }
         $var_name = ucwords(str_replace(array('$', '_'), array('', ' '), $var_name));
         $heading_tag = EEH_Debug_Tools::headingTag($heading_tag);
-        $result = EEH_Debug_Tools::headingSpacer($heading_tag);
-        $result .= EEH_Debug_Tools::heading($var_name, $heading_tag, $margin, $line);
+        // $result = EEH_Debug_Tools::headingSpacer($heading_tag);
```
Comment (manzoorwanijk) [inline core/helpers/EEH_Debug_Tools.helper.php:596]: may be remove this?

**Review by manzoorwanijk — APPROVED** [review]
LGTM

**Review by knazart — APPROVED** [review]
👌 changes.

File: core/helpers/EEH_Debug_Tools.helper.php (hunk 1 above)
Comment (tn3rb) [reply core/helpers/EEH_Debug_Tools.helper.php:596]: i might reinstate that in the  future so wanted to keep it

**Pull request merged**